src/SFA.DAS.Provider.PR.Web/Controllers/EmployerDetailsController.cs

```diff
@@ -0,0 +1,32 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.Encoding;
+using SFA.DAS.Provider.PR.Domain.Interfaces;
+using SFA.DAS.Provider.PR.Domain.OuterApi.Responses;
+using SFA.DAS.Provider.PR.Web.Authorization;
+using SFA.DAS.Provider.PR.Web.Infrastructure;
+using SFA.DAS.Provider.PR.Web.Models;
+
+namespace SFA.DAS.Provider.PR.Web.Controllers;
+
+[Authorize(Policy = nameof(PolicyNames.HasProviderAccount))]
+[Route("")]
+public class EmployerDetailsController(IOuterApiClient _outerApiclient, IEncodingService encodingService) : Controller
+{
+    [Route("{ukprn:int}/employers/{agreementid}", Name = RouteNames.EmployerDetails)]
```

Comment:
We should use `accountLegalEntityId` here to express the intent even though we know they both mean the same. The variable below can have suffix of `decoded`